src/bvh/iter.rs

```diff
@@ -33,7 +33,14 @@ impl<'bvh, 'shape, T: BHValue, const D: usize, Shape: Bounded<T, D>>
             stack: [0; 32],
             node_index: 0,
             stack_size: 0,
-            has_node: !bvh.nodes.is_empty(),
+            has_node: match bvh.nodes.first() {
+                // Only process the root leaf node if the shape's AABB is intersected.
+                Some(BvhNode::Leaf { shape_index, .. }) => {
+                    ray.intersects_aabb(&shapes[*shape_index].aabb())
+                }
+                Some(_) => true,
+                None => false,
+            },
```

Comment:
Please pull this out of the construction statement and just place it above and store the value in a `has_node` variable. Should make it a bit more readable. I don't like it when the construction itself has too much logic in it. The previous logic here was fine because it wasn't much code.